### ISSUE TITLE
Naming

### ISSUE THREAD

**Issue by Eisfunke** [hackmdio/codimd#1219]
Currently, there are two active repositories/softwares named "CodiMD": This one and [codimd/server](https://github.com/codimd/server). Both are actively maintained, but by different people. Especially for someone new to CodiMD/HackMD it's almost impossible to understand what's what.

As I understand, this repo is now maintained by you, the HackMD people, and you want to use it as open-core-model within HackMD Enterprise. The people who maintained this repo beforehand, who did the initial renaming to "CodiMD", are now at codimd/server.

What are your plans regarding the naming situation? I'd propose to rename this to HackMD Community Edition, because it's the conventional naming scheme for open-core-models, and it would make it far easier to understand that codimd/server is seperate and maintained by the people who created the name, and this and HackMD EE belong together and are maintained by the same people.

**Comment by GitOnion** [comment]
Thanks for raising this issue. Our plan is to continue using CodiMD. "CodiMD" is the name [discussed, approved, and changed **by the community here** at hackmdio/codimd](https://github.com/hackmdio/codimd/pull/850). We acknowledge the fact that some of the open source community forked our work and want to maintain their own version. In fact, there are 600+ forks and the one you mentioned is only one of them.
It is not conventional in the open source culture that the original repo changes its name every time someone forks it.

**Comment by ccoenen** [comment]
Linking #720 for completeness, where the discussion was.

I'm not particularly agreeing with your statement. Especially your portrayal that https://github.com/codimd/server would in any way be comparable to the 618 forks.

@GitOnion, you appear to have a background in economics. Would you kindly explain to me how the opportunity costs of this fight bring any value to HackMD? I don't really see what's in it for you. Especially when you could be using this time to make your actual product so much better? How does maintaining two separate brands make any sense at the point in developing your business?

**Comment by dragetd** [comment]
I also believe that both communities would benefit from this more clear separation. 

I would welcome if this change was made.

**Comment by Turakar** [comment]
Currently, I am wondering if I should switch to `codimd/server` in favour of `hackmdio/codimd`. For this decision, it is important to know which repository would be more suitable for a small-scale private server setup as mine. Therefore, I want to ask you some questions about this repo, also related to the naming decision:
1. Do you commit to support this repository in the future? (If yes, what is the reasoning behind that?)
2. Do you plan to refactor this repository as proposed in #1170 ? Do you plan to establish an CE/EE model?
3. Who should be the main contributors to this project? The HackMD team or solely the community?
These questions are currently unclear to me and I would be glad to know which way you want to go.

**Comment by hoijui** [comment]
So as I understand the history as an outsider (please correct me if I am wrong anywhere, a lot of it is just guessing):

1. HackMD started a new project and called it CodiMD (the software)
2. An other group of people made a fork of it, and they called themselves CodiMD (as a group)
3. Both projects are still very active
4. confusion arises among outsiders ("Which is my CodiMDs source?", "Where should I contribute?", ...)

I agree that this is a bad situation, but I can imagine a few ways of solving it:

1. re-unite the projects into one repo
2. rename one of the projects
3. clarify the relation of the two projects within the README's of the two repos, in a way it diffuses confusion of people (not sure if this is a practically viable solution)
4. do nothing, let confusion rise, until at some point, we might get a 3rd active fork with a different name, where people would flock to, and which would bring the slow death of both CodiMD repos

If _1._ is not an option, the common thing to do is definitely _2._. It is also clear that the original would have the entitlement to keep it's name, and the later fork would have to choose a new one.

With this as the -- in my eyes -- best of the likely options, things would normalize quickly.

Also spiritually, it is a great exercise: do not get attached too much to something, like a name. ;-) :P

**Comment by hoijui** [comment]
related issue on the other projects repo:
https://github.com/codimd/server/issues/204

**Comment by SISheogorath** [comment]
Hi @hoijui, the timeline is not entirely correct. At the CodiMD organization, soon after the split we wrote down the history in order to give people an understanding about it:

https://github.com/codimd/server/blob/master/docs/history.md

TL;DR:

1. HackMD started the project as HackMD.
2. The project became inactive, a bunch of contributors and I wondered why and asked if we could simply pick up the work from there and continue to free software version, while the HackMD team wanted to focus on their SaaS business only. https://github.com/hackmdio/codimd/issues/531
3. After a lot of confusion, we decided to come up with a new name https://github.com/hackmdio/codimd/issues/720#issuecomment-395245665 
4. After more than a year and us pushing the now named CodiMD project forward without large involvement from the HackMD team, we wanted to move the project into an own organization to simplify workflows and finally break the confusion about the two projects. Therefore I reached out to the HackMD team since the repository was still under the HackMD organization and I wouldn't just move a repository without talking the people who own the organization.
5. The HackMD team came up with a different idea, as alternative to our independence idea https://github.com/hackmdio/hackmd/issues/1170
6. At some point this debate came out of hand and we decided to hard-fork the project we worked on for more than a year in order to get independence even without the HackMD teams consent. https://github.com/codimd/server/issues/10
7. With the hard fork we asked the HackMD team, since they purposed to bring the SaaS product and the CodiMD project residing in this repository here together, to switch the name back to HackMD (as you can see in this issue, others had the same idea).
8. You wrote your comment here.

I agree that the situation is bad for external people. I agree that this needs to be resolved. But I hope you can imagine that I'm not about to give up a project name that I invented that easily. To me, the people who started the discussion, who participated in the discussion about the name and who finally made the decision along with a lot of other decisions in the development in the year without the HackMD team's involvement, are the Community and project members that this name defines. To me this name is not about the few lines of JS that run around in a repository, but about the people behind the project who put their free time into it in order to provide awesome free software to the world.

The HackMD team is part of this community simply for their work they did on the project, and I invite them once more to work with us, if they want. (You may notice, that since the beginning, I invited @jackycute to our new organization to keep the door open.) After the situation we had before, I'm not willing to jump back under the umbrella of the HackMD organization again.

**Comment by ccoenen** [comment]
In short: the people who were developing it at the time when we chose CodiMD as a name are mostly at https://github.com/codimd/server - which is why we are keeping the name. The people that were developing it when it was called HackMD - before CE or anything - are all here.

**Comment by hoijui** [comment]
uff... :D
I did not read the whole discussions I must admit, but even with your summary here plus [history.md](https://github.com/codimd/server/blob/master/docs/history.md), I feel I get the basic picture.
thank you!

If you changed the name, how much would you still care about having had to do so, one month later?

In live, there will always be (greater) powers standing in your way. They appear for you to learn to flow past without getting blocked. They dissolve into nothing, once you do so.

"Be water, my friend."

;-)

**Comment by TWille** [comment]
> uff... :D
> I did not read the whole discussions I must admit, but even with your summary here plus [history.md](https://github.com/codimd/server/blob/master/docs/history.md), I feel I get the basic picture.
> thank you!
> 
> If you changed the name, how much would you still care about having had to do so, one month later?
> 
> In live, there will always be (greater) powers standing in your way. They appear for you to learn to flow past without getting blocked. They dissolve into nothing, once you do so.
> 
> "Be water, my friend."
> 
> ;-)

Well there is always more to the story then just the name, including a whole initiative with domain and more to keep it alive.. I guess it is not that easy to change the name within the GitHub Organisation CodiMD ... I guess the easiest would be just renaming this one. How are the features synched? in both directions I guess?

**Comment by almereyda** [comment]
Hi Timm :wave:, features are not in any way synchronised between both CodiMDs.

The community organisation @codimd is a hard fork of the FLOSS originally released here.

**Comment by unteem** [comment]
@jackycute  could you be reasonable man and merge https://github.com/hackmdio/codimd/pull/1248
 and rename that repo HackMD-CE or whatever. That would make everyone ones life easier. 
We all know codimd's community is now gone living its own life. So please give it some space and to your hackmd community. Its for the best of both. Thanks

**Comment by jackycute** [comment]
From the timeline wise, it totally makes sense that we can use the name “CodiMD”. Take other open source project for example, it’s not practical the fork asks the original repo to change the name.

From the contribution wise, you said that you came up the name first and you took lots of time to run the brand and build the community, so you should have the name.

Then, I started this project from 2015, and I spent almost one and half year working on this on my own. I should have refused you to change the name of the repo at first place - but I didn’t. I should have rejected you forking to another org and stop you telling people to go to your org - but I didn’t.

That’s because I’m open-minded, I contribute to “this” project for open source and I believe everyone here has the same mindset to contribute to this project.

So, stop telling us to change the name, you are standing on your side and speaking your own words. We can decide our project's name.

**Comment by dragetd** [comment]
This is rather disappointing. :(

**Comment by pierreozoux** [comment]
At first, I was really happy with HackMD team.
With the open core announcement, I was really happy they could make a living from it.
Then I realised it was not open core, I was a but disappointed but I understood that it was just some misunderstanding. I proposed to change the name to make it clear.
After the discussion , my feeling about hackmd team was neutral plus.
Then the discussion about the fork, I was neutral.

Now I see this.. I'm really really angry. And I'll not say more, I don't want to right now, I'll regret it later.

@jackycute we know you have this free software heart, please reconsider this, it would be so much easier for everyone!

Thanks again for what you started, and I hope it will be a happy ending.

**Comment by dragetd** [comment]
For everyone following this issue: The pig-headed attitude of the hackmd team did never change. Multiple attempts to sort things out failed.

The codimd community decided to no longer put energy into this and and improve the situation for users who might be confused about the name by now renaming codimd (again) to HedgeDoc. The migration will take place and move things from https://github.com/codimd to https://github.com/hedgedoc .

It is a real shame that even after over a year of trying to reach out, things had to end like this. As a user and IT professional, I will make sure to stay away from the people involved in hackmd in my work.

**Comment by jackycute** [comment]
I've clearly stated we have no plan to change the name at the moment and closed this issue. There is no responsibility for us to do this and we also are not the one who caused this, don't try to misleading the thread.
